### PR TITLE
DataFrame.get_rows should take Column, not Sequence

### LINKED ISSUE #146: DataFrame.get_rows should take Column, not Sequence
xref https://github.com/data-apis/dataframe-api/pull/140#discussion_r1170289956

I agree, and came across this in my seaborn POC, where I have to convert a Column to a sequence to be able to pass it in

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #146

### CODE REVIEW

File: spec/API_specification/dataframe_api/dataframe_object.py [hunk 1]
```diff
@@ -88,24 +88,18 @@ def get_columns_by_name(self, names: Sequence[str], /) -> DataFrame:
         """
         ...
 
-    def get_rows(self, indices: Sequence[int]) -> DataFrame:
+    def get_rows(self, indices: Column[int]) -> DataFrame:
```
Comment (kkraus14) [inline spec/API_specification/dataframe_api/dataframe_object.py:91]: is `Column[int]` valid typing?

File: spec/API_specification/dataframe_api/dataframe_object.py [hunk 2]
```diff
@@ -88,24 +88,18 @@ def get_columns_by_name(self, names: Sequence[str], /) -> DataFrame:
...
 
-    def get_rows(self, indices: Sequence[int]) -> DataFrame:
+    def get_rows(self, indices: "Column[int]") -> DataFrame:
         """
         Select a subset of rows, similar to `ndarray.take`.
 
         Parameters
         ----------
-        indices : Sequence[int]
+        indices : Column
```
Comment (MarcoGorelli) [inline spec/API_specification/dataframe_api/dataframe_object.py:97]:
```suggestion
        indices : Column[int]
```

File: spec/API_specification/dataframe_api/dataframe_object.py (hunk 1 above)
Comment (MarcoGorelli) [reply spec/API_specification/dataframe_api/dataframe_object.py]: i've put quotes around it - `Column[bool]` is used in some other places, I just took it from there

**Review by kkraus14 — APPROVED** [review]

**Review by rgommers — APPROVED** [review]
Seems like we're all good on this one. LGTM too, so in it goes.

**Pull request merged**